### PR TITLE
Clean up error handling

### PR DESCRIPTION
**Pull request by ywwg**
All <500 errors should only be info-logged because they are expected and not a problem for the operation of the server.  >=500s are warnings because they are trouble, but do not represent unrecoverable server errors.

Note that we are returning error contents into the http error.  We should make sure that there's no chance that internal error details might get returned to clients.  I don't think that is a possibility but we should spot-check as development continues to make sure

### CODE REVIEW

File: pkg/influx/parser.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func influxPointToTimeseries(pt models.Point) ([]cortexpb.TimeSeries, error) {
 
 	fields, err := pt.Fields()
 	if err != nil {
-		return nil, fmt.Errorf("error getting fields from point: %w", err)
+		return nil, errorx.Internal{Msg: "error getting fields from point", Err: err}
```
Comment (ywwg) [inline pkg/influx/parser.go:76]: I am not exactly sure what kind of error this is, but it seems like an internal, not a client error.

File: pkg/influx/errors.go [hunk 2]
```diff
@@ -26,42 +21,50 @@ func tryUnwrap(err error) error {
...
 	case errors.Is(err, context.DeadlineExceeded) || isGRPCTimeout(err):
+		httpErrString = "network timeout"
 		statusCode = http.StatusGatewayTimeout
-		level.Error(logger).Log("msg", "response timeout", "err", err)
+	case errors.Is(err, context.Canceled):
+		// Note: It seems unlikely this can happen other than as a timeout, so we
+		// should call it an internal error.
+		httpErrString = "request cancelled"
+		statusCode = http.StatusInternalServerError
 	case isNetworkTimeout(err):
```
Comment (carrieedwards) [inline pkg/influx/errors.go:46]: The isGrpcTimeout method at the bottom of the file can be deleted

Comment (ywwg) [reply pkg/influx/errors.go:43]: done

Comment (ywwg) [reply pkg/influx/errors.go:46]: I restored references to this function so we do need it again

**Review by carrieedwards — APPROVED** [review]

**Pull request merged**